datasync/chaindatafetcher/api.go

```diff
@@ -49,7 +49,11 @@ func (api *PublicChainDataFetcherAPI) StartRangeFetching(start, end uint64, reqT
 	case "trace":
 		t = types.RequestTypeTraceGroup
 	default:
-		return errors.New("the request type should be 'all', 'block', or 'trace'")
+		ut, ok := reqType.(float64)
+		if !ok {
+			return errors.New("the request type should be 'all', 'block', 'trace', or uint type")
+		}
+		t = types.RequestType(ut)
```

Comment:
@jeongkyun-oh Don't we need to check the valid range of uint value?

Reply:
@kjhman21 Added validation logic for request type.